### DISCUSSION TITLE
My Complete Obsidian Workflow to Manage My Life - Mathis Gauthey's Website

### DISCUSSION THREAD

**Discussion by giscus** [mathisgauthey/mathisgauthey.github.io#3] [_]
# My Complete Obsidian Workflow to Manage My Life - Mathis Gauthey's Website

My Complete Obsidian Workflow to Manage My Life Introduction Why Writing This Guide ? As I promised to quite a few people on different places, I&rsquo;m releasing a full blown guide of my Obsidian workflow. It has many cool features that I&rsquo;ve seen people struggling with online :
Tasks and project management with journaling, workout logging and media tracking. Using daily and periodic notes. Read-it-later with automated conversion to markdown and import inside your vault.

https://mathisgauthey.github.io/my-complete-obsidian-workflow-to-manage-my-life/

**Comment by 12risobo** [comment]
Hi Mathis, I have been using Obsidian for 2 years and it's great to see how you organized your vault. There is so much useful information in your post and downloadable vault. Big thanks for sharing this!

**Reply by mathisgauthey** [reply]
Hey there, thanks you for your comment it means a lot. 

I certainly hope I'll help as many people as possible with these guides and will continue posting some when I have other things to share !

**Comment by rolanddeboer** [comment]
Hey, this is really useful and inspiring. Thank you for sharing this.

**Reply by mathisgauthey** [reply]
Thanks for your comment ✌️

**Comment by incarnator1378** [comment]
Absolutly great!

**Reply by mathisgauthey** [reply]
Thanks ✌️

**Reply by incarnator1378** [reply]
Hi, bro! 
Need u help about Reccuring tasks: 
1. Area Projects, My First Project, i add task "What i need to do?"
2. In task menu: Recurs - every day, Due - 2023-08-30, apply
3. In today daily note scroll to "Recurring" - 0 task
Maybe i do something wrong?

**Reply by mathisgauthey** [reply]
Hi there ! It could be two things :
- Either you created the recurring task directly inside an `Area` file under the `2-Areas` folder. You can't do that, tasks inside `Areas` are projects, could link to a project note inside `1-Projects` and create the task here.
- Maybe you didn't set a first due date. Recurring tasks need a due date to work the recurrence on. I'll refer you to the task documentation.

With that and the example vault that features a recurring tasks that works, you should be able to sort things out !

Feel free to tell me if it worked or not, I'll give it a closer look.

**Reply by incarnator1378** [reply]
Oh, so simple, got it, thnx!

**Comment by bribrigugu** [comment]
Hello Mathis, thank you very much for sharing your work with thorough explanation. I am just starting out with Obsidian and found your workflow fitting my needs. I followed your guide word by word, however when adding a new note based on template (Daily note, monthly note...), the title is always set to "Invalid date". Do you know how to correct this? Sorry if this question seems dumb but I am very new to all this. Thank you!

**Reply by bribrigugu** [reply]
Sorry to ask again but I have been trying to search for the solution online but I don't understand "code". It seems that Moment might be the issue but I don't know how to fix it.
Also another problem I have is that when I create a new note from template, the year is 0001 instead of 2023... I would be really grateful if you can help with this issue. Thank you very much.

**Reply by mathisgauthey** [reply]
Hey there. I'm not sur what could be wrong there, have you tried starting again from scratch using my template ? I don't see any other issue like that on the comment thread and am not sure what could be wrong.

I'd recommend making sure your plugins are activated in the `Community Plugins` section as well.

**Reply by bribrigugu** [reply]
Thanks for replying. Yes I tried starting it all over and checking the Community Plugins but the problem is still here. Thank you for answering anyway.

**Reply by mathisgauthey** [reply]
Can you confirm that the top of your `Daily-Notes` template file is using the following format :

```
---
title: <% moment(tp.file.title, "YYYY-MM-DD").format("YYYY-MM-DD") %>
date: <% moment(tp.file.creation_date()).format("YYYY-MM-DD HH:mm:ss") %>
lastmod: <% moment(tp.file.creation_date()).format("YYYY-MM-DD HH:mm:ss") %>
categories: 
tags: daily-notes
aliases: 
share: false 
---

# <% moment(tp.file.title, "YYYY-MM-DD").format("YYYY-MM-DD") %>
```

If so, I'd recommend playing around `Templater `settings and `filename-heading-sync` plugin.

**Reply by bribrigugu** [reply]
Hello, yes this is how it looks like. I will try to play around and see if it changes anything and let you know. Thank you very much for replying.

**Reply by bribrigugu** [reply]
By the way, the "Invalid date" issue goes away when I create my second Daily note. However, the year of the second note is 0001 instead of 2023. The third note's year is 0002, fourth is 0003... I will try to search where does it come from and let you know if I find out.

**Reply by bribrigugu** [reply]
Hello again Mathis, I think I resolved the issue (please note that I don't code and don't really know what I am doing, just trying and testing...).

For some reason, the "YYYY-MM-DD" doesn't give the real year. So I changed the Daily note template to this (if you have an explanation, I would be really interested to understand) : 

---
title: <% moment(tp.file.creation_date()).format("YYYY-MM-DD") %>
date: <% moment(tp.file.creation_date()).format("YYYY-MM-DD HH:mm:ss") %>
lastmod: <% moment(tp.file.creation_date()).format("YYYY-MM-DD HH:mm:ss") %>
categories: 
tags: daily-notes
aliases: 
share: false 
---

# <% moment(tp.file.creation_date()).format("YYYY-MM-DD") %>

Hi partner, today is <% moment(tp.file.creation_date()).format("dddd") %>.

<%*
const currentMoment = moment(tp.file.creation_date());
const hash = '# ';
const slash = ' / ';
const pipe = ' | ';
const leftAngle = '❮ ';
const rightAngle = ' ❯';
tR += leftAngle;
tR += '[[' + currentMoment.format("YYYY") + ']]' + slash;
tR += '[[' + currentMoment.format("YYYY-[Q]Q|[Q]Q") + ']]' + slash;
tR += '[[' + currentMoment.format("YYYY-MM|MMMM") + ']]' + slash;
tR += '[[' + currentMoment.format("gggg-[W]ww") + '|' + currentMoment.format("[Week] ww") + ']]';
tR += rightAngle;
tR += '\n';
tR += '\n';
tR += leftAngle;
currentMoment.add(-1,'days');
tR += '[[' + currentMoment.format("YYYY-MM-DD") + ']]' + pipe;
currentMoment.add(1,'days');
tR += currentMoment.format("YYYY-MM-DD") + pipe;
currentMoment.add(1,'days');
tR += '[[' + currentMoment.format("YYYY-MM-DD") + ']]';
currentMoment.add(-1,'days');
tR += rightAngle;
%>

**Reply by mathisgauthey** [reply]
Just to be sure, where is the issue when you create the note ? The filename or the first heading ?

And are you creating it using the periodic note plugin command / ribbon icon ?

**Reply by bribrigugu** [reply]
Again thank you for taking the time to reply and explain! The issue is with the title, which is reproduced in the heading, the first line of texte and the rest of the note. And I am creating them using the periodic note plugin command.
By the way, after playing around with different plugins, my current solution is : deactivating filename-heading-sync plugin and the problem disappears.
I need to learn all of this more in depth but in the meantime, thank you very much because your vault is great and you have been very helpful.

**Reply by mathisgauthey** [reply]
Alright, yes indeed `filename-heading-sync` might copy the first header to the heading on file creation before the periodic plugin changes the title to a date, or something like that. It was quite a pain to configure back there, but I don't understand why it doesn't work with your case, it is the first report of such an issue yet so if you find something feel free to share it as I can't seem to reproduce the issue myself.

Here you can find my `filename-heading-sync` plugin config :

![Screenshot_20231007-082954.png](https://github.com/mathisgauthey/mathisgauthey.github.io/assets/46576952/435bbe6c-1055-4a96-8c98-98a1a265777a)

But it might as well be an issue with periodic note not changing the `title` from `Daily-Notes` to `YYYY-MM-DD`, maybe induced by `filename-heading-sync` ? Well, something like that.

**Comment by kwetiaw-goreng** [comment]
hey, great vault, im also just starting with obsidian. 
I stumbled to this as im looking for a workout tracker, i cloned your vault but i cant seem to find any workout template? do i need to do something to enable it?
thanks!

**Reply by mathisgauthey** [reply]
Hey there ! Sure, you can find the basic workout logging on the [0.0.1](https://github.com/mathisgauthey/obsidian-workflow-template/releases/tag/0.0.1) version. It was then removed in the [1.0.0](https://github.com/mathisgauthey/obsidian-workflow-template/releases/tag/1.0.0) because Obsidian mobile was too slow and I discovered Hevy which is objectively a better way to get exercises progress over time without the need to open past workout notes ✌️

**Comment by moldyCr0w** [comment]
This vault is great. Very comprehensive and not overwhelming (I've seen some others that are...). This is excellent.

Couple of questions:
1. I'm guessing I don't need a pro version of todoist since the projects in todoist won't sync with my projects in Obsidian, correct? This is basically to capture tasks which I can then clarify, organize, review, and engage. Correct?
2. Some of your boards have been deprecated, I think? The Taskboard, for example? Just checking that I'm not missing something.
3. This may seem stupid but: how do I turn a task into a project in Obsidian? Like, I have the checkbox there in my Daily-Notes; now I want to make that checkbox a project. Is the workflow to just create a project of the samename and check the box? And therefore the task should be "Create Project with this name?" Sorry if this seems dumb...


Thanks for this. I'm still wrapping my head around using todoist with Obsidian...I'm not sure that I see the advantage except for the Inbox capturing things and then somehow making them Projects under Projects or Areas...

**Reply by mathisgauthey** [reply]
Heyt here, thanks for the compliment, glad it helps !

I'll try to answer :
1. Yup you're right, this is just a better Inbox. Obsidian is quite slow to open on mobile, hence the need for an external service !
2. I think a combination of Obsidian updates and plugins updates broke a few things here and there, I'd be glad to review pull request on that matter if someone fixes it and share it with the world !
3. A project is a task that contains a related note. For instance : `- [ ] Do my [[table wood working project]]` would be considered a project. And you can add subtasks inside the `table wood working project` note.

Do as you feel, it's just a starting point, now you'll need to figure out what works for you ! I use Todoist exclusively as an Inbox, some might use it for more or don't use it at all o/
Have a great day!
